### ISSUE TITLE
AssertionFailedException: All jumps into the loop body should go through the entry point

### ISSUE THREAD

**Issue by greenozon** [icsharpcode/ILSpy#1558]
ILSpy version 5.0.0.4850-preview2

```
 ---> ICSharpCode.ILSpy.AssertionFailedException: All jumps into the loop body should go through the entry point
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message, String detailMessage) in offset 380
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message) in offset 9
   at System.Diagnostics.TraceInternal.Fail(String message) in offset 55
   at ICSharpCode.Decompiler.IL.ControlFlow.LoopDetection.IncludeUnreachablePredecessors(List`1 loop) in offset 94
   at ICSharpCode.Decompiler.IL.ControlFlow.LoopDetection.DetectSwitchBody(Block block, SwitchInstruction switchInst) in offset 348
   at ICSharpCode.Decompiler.IL.ControlFlow.LoopDetection.Run(Block block, BlockTransformContext context) in offset 88
   at ICSharpCode.Decompiler.IL.Block.RunTransforms(IEnumerable`1 transforms, BlockTransformContext context) in offset 69
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 146
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 89
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 89
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 89
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.Run(ILFunction function, ILTransformContext context) in offset 146
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 595
-- continuing with outer exception (ICSharpCode.Decompiler.DecompilerException) --
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 1083
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DoDecompile(IMethod method, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 249
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(IEnumerable`1 definitions) in offset 412
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(EntityHandle[] definitions) in offset 1
   at ICSharpCode.ILSpy.CSharpLanguage.DecompileMethod(IMethod method, ITextOutput output, DecompilationOptions options) in offset 246
   at ICSharpCode.ILSpy.TreeNodes.MethodTreeNode.Decompile(Language language, ITextOutput output, DecompilationOptions options) in offset 15
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.DecompileNodes(DecompilationContext context, ITextOutput textOutput) in offset 69
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.<>c__DisplayClass34_0.<DecompileAsync>b__0() in offset 38
```

https://www.sendspace.com/file/wgfym2

**Comment by siegfriedpammer** [comment]
```csharp
// Wizard2004.PolygonSeries
[MethodImpl(MethodImplOptions.NoInlining)]
public PolygonSeries()
{
	while (false)
	{
		((object[])null)[0];
	}
	uPmRCyK7uWQraaXwfCTk();
	num = (IFj44qK7aM4SerPRIsCF() ? 3 : 4);
	while (true)
	{
		switch (num)
		{
		case 5:
			return;
		case 2:
		{
			label = DefaultSeriesLabel();
			selected = false;
			num = 5;
			break;
		}
		case 1:
		case 3:
		{
			base..ctor();
			goto case 2;
		}
		default:
		{
			num = 2;
			break;
		}
		case 0:
		case 4:
		{
			OooemCTOZuseaWhuJLk.YDjOpM0zWqPfX();
			goto case 1;
		}
		}
	}
}
```

**Comment by tamlin-mike** [comment]
Perhaps run the obfuscated assembly through de4dot first; It's known to fix things up enough for ilSpy to work.

**Comment by dgrunwald** [comment]
See also: #1242

**Comment by dgrunwald** [comment]
Fixed in 8117dfff4a2b238a0e2794416c78b9d3656dbfb8.
